app/views/Settings.js

```diff
@@ -53,11 +57,49 @@ export const SettingsScreen = ({ navigation }) => {
     };
   }, []);
 
+  // on load setting page, this method will check whether we have remove location permission from app settings.
+  // It will set status "Location Active" only if permission is granted./*tested properly on android device*/
+  const willParticipate = () => {
+    SetStoreData(PARTICIPATE, 'true').then(() => {
+      // Turn of bluetooth for v1
+      //BroadcastingServices.start();
+    });
+    // Check and see if they actually authorized in the system dialog.
+    // If not, stop services and set the state to !isLogging
+    // Fixes tripleblindmarket/private-kit#129
+    BackgroundGeolocation.checkStatus(({ authorization }) => {
+      if (authorization === BackgroundGeolocation.AUTHORIZED) {
+        LocationServices.start();
+        setBGLocationLogging(true);
+        setIsLogging(true);
```

Comment:
It looks like you are trying to track these as two separate booleans:

- `isLogging` = user PARTICIPATE state
- `isBgLogging` = actual bg service state

if that's the case, then don't set isLogging to true here, because now you've muddled the state between the two variables.

Reply:
Once you use the new checkStatus, this code should become more clear.